docs/source/setup/contributing.rst

```diff
@@ -0,0 +1,62 @@
+Contributing
+============
+
+Conversations about development and issues take place in the `GitHub repository <https://github.com/N-Wouda/ALNS/>`_.
+Feel free to open a new issue if you have something to discuss.
+
+
+Setting up a local installation
+-------------------------------
+
+First, fork the ``alns`` repository from the `GitHub website <https://github.com/N-Wouda/ALNS/fork>`_.
+Then, clone your new fork to some local environment:
+
+.. code-block:: shell
+
+   git clone https://github.com/<your username>/ALNS.git
+
+Follow the rest of the instructions in :ref:`running-locally` on setting up ``poetry``, and installing the required dependencies.
+This might take a few minutes.
+
+Now make sure everything runs smoothly, by executing the test suite:
+
+.. code-block:: shell
+
+   poetry run pytest
+
+If you use `pre-commit <https://pre-commit.com/>`_, you can use our pre-commit configuration file to set that up too.
+Simply type:
+
+.. code-block:: shell
+
+   pre-commit install
+
+After this completes, comming style and typing issues are automatically checked whenever you make a new commit to your feature branch.
+Note that we do not use pre-commit via ``poetry``.
```

Comment:
> We do have `pre-commit` as poetry dependency, so what do we mean that pre-commit is not used via `poetry`?

Only for the CI. For local development we kind of rely on people having a global/system-wide `user-commit` (because git doesn't know about poetry anyway). But I'll just remove this, and leave it at "this thing exists and you can use it if you want".